nautobot_golden_config/jobs.py

```diff
@@ -309,31 +334,70 @@ class Meta:
-            )
-        if constant.ENABLE_COMPLIANCE:
-            config_compliance(
-                self.job_result,
-                self.logger.getEffectiveLevel(),
-                self.qs,
-                self.device_to_settings_map,
-            )
+        self.logger.debug("Compiling device data for GC job.", extra={"grouping": "Get Job Filter"})
+        self.qs = get_job_filter(data)
+        self.logger.debug(
+            f"In scope device count for this job: {self.qs.count()}", extra={"grouping": "Get Job Filter"}
+        )
+        self.logger.debug("Mapping device(s) to GC Settings.", extra={"grouping": "Device to Settings Map"})
+        self.device_to_settings_map = get_device_to_settings_map(queryset=self.qs)
+        gitrepo_types = ["backup_repository", "jinja_repository", "intended_repository"]
+        self.logger.debug(
+            f"Repository types to sync: {', '.join(sorted(gitrepo_types))}",
+            extra={"grouping": "GC Repo Syncs"},
+        )
+        current_repos = get_refreshed_repos(job_obj=self, repo_types=gitrepo_types, data=self.qs)
+        failed_jobs = []
+        error_msg = ""
+        for enabled, play in [
+            (constant.ENABLE_INTENDED, config_intended),
+            (constant.ENABLE_BACKUP, config_backup),
+            (constant.ENABLE_COMPLIANCE, config_compliance),
+        ]:
+            try:
+                if enabled:
+                    play(self)
+            except BackupFailure:
+                self.logger.error("Backup failure occurred!")
+                failed_jobs.append("Backup")
+            except IntendedGenerationFailure:
+                self.logger.error("Intended failure occurred!")
+                failed_jobs.append("Intended")
+            except ComplianceFailure:
+                self.logger.error("Compliance failure occurred!")
+                failed_jobs.append("Compliance")
+            except Exception as error:  # pylint: disable=broad-exception-caught
+                error_msg = f"`E3001:` General Exception handler, original error message ```{error}```"
+        now = make_aware(datetime.now())
+        self.logger.debug(
+            f"Finished the {self.Meta.name} job execution.",
+            extra={"grouping": "GC After Run"},
+        )
+        if current_repos:
+            for _, repo in current_repos.items():
+                if repo["to_commit"]:
+                    self.logger.debug(
+                        f"Pushing {self.Meta.name} results to repo {repo['repo_obj'].base_url}.",
+                        extra={"grouping": "GC Repo Commit and Push"},
+                    )
+                    repo["repo_obj"].commit_with_added(f"{self.Meta.name.upper()} JOB {now}")
+                    repo["repo_obj"].push()
+        if len(failed_jobs) > 1:
+            failed_jobs = ", ".join(failed_jobs)
+        elif len(failed_jobs) == 1:
+            failed_jobs = failed_jobs[0]
+        elif len(failed_jobs) == 0:
```

Comment:
same as the other job, this causes a failure to be logged even if no jobs failed.